org.mixedrealitytoolkit.input/InteractionModes/NearInteractionModeDetector.cs

```diff
@@ -20,10 +20,57 @@ public class NearInteractionModeDetector : ProximityDetector
         [Tooltip("The set of near interactors that belongs to near interaction")]
         private List<XRBaseInteractor> nearInteractors;
 
+        /// <summary>
+        /// Used to keep track of the previously detected colliders so that we can know which
+        /// colliders stopped being detected and update their buttons front plate and rounded
+        /// rect if they are labeled as dynamic (based on proximity).
+        /// </summary>
+        List<Collider> previouslyDetectedColliders = new List<Collider>();
+
         /// <inheritdoc />
         public override bool IsModeDetected()
         {
-            return base.IsModeDetected() || IsNearInteractorSelecting();
+            bool result = base.IsModeDetected() || IsNearInteractorSelecting();
+            if (result)
+            {
+                for (int i = 0; i < previouslyDetectedColliders.Count; i++)
+                {
+                    Collider previouslyDetectedCollider = previouslyDetectedColliders[i];
+                    if (!DetectedColliders.Contains(previouslyDetectedCollider) && previouslyDetectedCollider != null)
+                    {
+                        INearInteractionModeTarget nearInteractionMode = previouslyDetectedCollider.GetComponent<INearInteractionModeTarget>();
+                        if (nearInteractionMode != null)
+                        {
+                            foreach (XRBaseInteractor xrBaseInteractor in nearInteractors)
+                            {
+                                previouslyDetectedCollider.GetComponent<INearInteractionModeTarget>().UnregisterActiveColliderWithInteractor(previouslyDetectedCollider, xrBaseInteractor);
+                            }
+                            if (previouslyDetectedCollider.GetComponent<INearInteractionModeTarget>().ActiveColliderWithInteractorCount == 0)
```

Comment:
I would like some other opinions here (maybe @keveleigh, @david-c-kline , @whebertML , @ghazen-ml , @shaynie )....

But I think we can safely push the `SetEnableDynamicComponents` out of the `INearInteractionModeTarget` interface, and force the interface implementation to call `SetEnableDynamicComponents` itself.   That is, I'm in favor of a simpler interface.

Reply:
I cannot do that (I tried though).  If I remove them from the interface then the NeartInteractionModeDetector is unable to call them directly from the PressableButton instance because that would require to NearInteractionModeDetector to know PressableButton type and that would generate a circular reference between packages.

Reply:
As per offline discussion, I misunderstood the original suggestion as just removing the methods from the interface.  I have removed the methods from the interface and moved the logic to update the state to PressableButton class.  Resolving this.